compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -5700,7 +5700,8 @@ private BErrorType createErrorType(BType lhsType, BType rhsType, BType detailTyp
 
     public BErrorType createErrorType(BType detailType, long flags, SymbolEnv env) {
         String name = anonymousModelHelper.getNextAnonymousIntersectionErrorTypeName(env.enclPkg.packageID);
-        BErrorTypeSymbol errorTypeSymbol = Symbols.createErrorSymbol(flags | Flags.ANONYMOUS, names.fromString(name),
+        BErrorTypeSymbol errorTypeSymbol = Symbols.createErrorSymbol(flags | Flags.ANONYMOUS | Flags.PUBLIC,
```

Comment:
It's better to extract the flags to a separate variable. In line 5709, we set the flags of the error type again from the type symbol. By extracting it to a variable, we can directly set it there as well.

Reply:
Within that function, we do the following kind of check:
```java
if (typeDef.flagSet.contains(Flag.PUBLIC)) {
            flags = Flags.PUBLIC;
        }
```
Here, we only add the `public` flag if the `typedef` contains the `public` flag. In that instance, is it correct to add the `public` flag to every intersection error type that we create?
I don't think we need to change this path. We only need to change the path where we create an intersection from two errors, which calls this function from `createErrorType(BType lhsType, BType rhsType, BType detailType, SymbolEnv env)`.